aptos-move/aptos-vm/src/validator_txns/dkg.rs

```
@@ -5,132 +5,108 @@ use crate::{
-    on_chain_config::{ConfigurationResource, OnChainConfig},
-    transaction::{ExecutionStatus, TransactionStatus},
+    transaction::ExecutionStatus,
+    validator_txn::ValidatorTransaction,
 };
 use aptos_vm_logging::log_schema::AdapterLogSchema;
 use aptos_vm_types::output::VMOutput;
 use move_core_types::{
     account_address::AccountAddress,
     value::{serialize_values, MoveValue},
-    vm_status::{AbortLocation, StatusCode, VMStatus},
+    vm_status::VMStatus,
 };
 use move_vm_runtime::module_traversal::{TraversalContext, TraversalStorage};
 use move_vm_types::gas::UnmeteredGasMeter;
 
-#[derive(Debug)]
-enum ExpectedFailure {
-    // Move equivalent: `errors::invalid_argument(*)`
-    EpochNotCurrent = 0x10001,
-    TranscriptDeserializationFailed = 0x10002,
-    TranscriptVerificationFailed = 0x10003,
-
-    // Move equivalent: `errors::invalid_state(*)`
-    MissingResourceDKGState = 0x30001,
-    MissingResourceInprogressDKGSession = 0x30002,
-    MissingResourceConfiguration = 0x30003,
-}
-
-enum ExecutionFailure {
-    Expected(ExpectedFailure),
-    Unexpected(VMStatus),
-}
-
 impl AptosVM {
-    pub(crate) fn process_dkg_result(
+    pub(crate) fn process_dkg_transaction(
         &self,
         resolver: &impl AptosMoveResolver,
         log_context: &AdapterLogSchema,
         session_id: SessionId,
-        dkg_transcript: DKGTranscript,
+        dkg_transaction_data: DKGTransactionData,
     ) -> Result<(VMStatus, VMOutput), VMStatus> {
-        match self.process_dkg_result_inner(resolver, log_context, session_id, dkg_transcript) {
+        match self.process_dkg_transaction_inner(
+            resolver,
+            log_context,
+            session_id,
+            dkg_transaction_data,
+        ) {
             Ok((vm_status, vm_output)) => Ok((vm_status, vm_output)),
-            Err(Expected(failure)) => {
-                // Pretend we are inside Move, and expected failures are like Move aborts.
```

We need to do this too. The VM should return an error status code for the transaction but continue operation when an expected failure occurs. If the VM emits an actual `Err` then execution of the whole block will fail and the consensus will panic. Are you catching the new errors anywhere else? Perhaps I've not got there yet.

---

fixed